contracts/eip712.cairo

```diff
@@ -46,23 +44,111 @@ func add_prefix{range_check_ptr, bitwise_ptr : BitwiseBuiltin*}(value : felt, pr
     return (result, overflow)
 end
 
-func get_hash{range_check_ptr, bitwise_ptr : BitwiseBuiltin*}(
-        to : felt, selector : felt, calldata_len : felt, calldata : felt*, nonce : felt, domain_hash : Uint256) -> (
-        hashed_msg : Uint256):
+func encode_call{range_check_ptr, bitwise_ptr : BitwiseBuiltin*}(
+    values: Uint256*,
+    to : felt, selector : felt, calldata_len : felt, calldata : felt*
+) -> ():
     alloc_locals
-    let (calldata_uint256) = map_to_uint256(calldata, calldata_len)
-    let (calldata_hash) = uint256_keccak(calldata_uint256, calldata_len * 32)
 
-    let (nonce_h, nonce_l) = split_felt(nonce)
     let (to_h, to_l) = split_felt(to)
+    assert values[0] = Uint256(to_l, to_h)
+
     let (selector_h, selector_l) = split_felt(selector)
+    assert values[1] = Uint256(selector_l, selector_h)
+
+    let (calldata_uint256) = map_to_uint256(calldata, calldata_len)
+    let (calldata_hash) = uint256_keccak(calldata_uint256, calldata_len * 32)
+    assert values[2] = Uint256(calldata_hash.low, calldata_hash.high)
+
+    return ()
+end
+
+func encode_calls_loop{range_check_ptr, bitwise_ptr : BitwiseBuiltin*}(
+    values: Uint256*,
+    call_index: felt,
+    call_array_len: felt,
+    call_array: AccountCallArray*,
+    calldata: felt*,
+) -> ():
+   if call_index == call_array_len:
+        return ()
+   end
+
+   let call_info = call_array[call_index]
+   let call_calldata = calldata + call_info.data_offset
+   let values = values + call_index * 6 # every call creates 3 uint256s, every uint256 = 2 felts
```

Comment:
I think it should be just `let values = values + 6`, as we pass values with offset down the loop, and using call_index would work if values pointer remained untouched.